design-docs/2021/q1/plugin-redesign.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Plugin Design Doc
+author: Kevin Siegler (@topocount)
+created: 2021-02-02
+---
+
+# Plugin Design Doc
+
+## Overview
+
+SourceCred plugins are almost completely beholden to their datasources.
+A local cache exists, but its only real purpose is to provide a local source of data for graph creation.
+Plugin caches cannot stand by themselves for too long and are easily blown out by upstream changes.
+Additionally, caches are only built in environments local to graph creation.
+They cannot easily be shared over a network, or provide any real source of truth outside of the API.
+
+@hammadj has designed an new [plugin model] that relies upon a `DataProvider` and `DataArchive` as a persistent mirror of data from a specific platform.
+This is a significant improvement over the existing model described above because it allows for the persistence of cred data independent of a platform ability (or willingness) to provide the raw information needed to build cred data over time.
+![snapshot of plugin model]
+
+[plugin model]: https://lucid.app/lucidchart/591bf744-fdbe-4aa2-a0c5-4be90f436a71/view?page=0_0#
+[snapshot of plugin model]: https://cdn.discordapp.com/attachments/714955718788382750/806248646625984542/Blank_diagram_-_Page_1.png
+
+## Informal Specification
```

Comment:
Should add a link to the HackMD as well: https://hackmd.io/@nnpZD6M7TlicIVX_NaLGlA/Sk_Ye8oxO/edit